### PR TITLE
feat: add user command

### PR DESCRIPTION
**Pull request by ppiques**

### CODE REVIEW

File: include/IRCPeer.hpp [hunk 1]
```diff
@@ -8,6 +8,9 @@
 # include <sys/socket.h>
 # include <unistd.h>
 # include "TCPPeer.hpp"
+# include "commands.hpp"
+
+class IRCMessage;
```
Comment (vfurmane) [inline include/IRCPeer.hpp:13]: Useless definition

File: src/commands/user.cpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#include "IRCErrors.hpp"
+
+void	command_user(IRCMessage &message, Dependencies &deps)
+{
+	if (message.getArgsCount() < 4)
+		throw ERR_NEEDMOREPARAMS(message.getCommand());
+	if (message.peer.isRegistered())
+		  throw ERR_ALREADYREGISTERED();
+	std::size_t pos;
+	int mode = stoi(message.getArguments[1], &pos, 10);
```
Comment (vfurmane) [inline src/commands/user.cpp:11]: Nice try but it is C++11

**Review by vfurmane — CHANGES_REQUESTED** [review]
meme so funny to descendre ppik gratuitement

File: test/TCP/Server.test.cpp [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#include "catch.hpp"
+#include <cstdio>
+
+#define private public
+#include "TCP/Server.hpp"
```
Comment (theovgl) [inline test/TCP/Server.test.cpp:5]: Hop ca saute

**Review by theovgl — CHANGES_REQUESTED** [review]

**Review by theovgl — APPROVED** [review]

**Review by vfurmane — APPROVED** [review]

**Pull request merged**